packages/cozy-authentication/src/Authentication.jsx

```diff
@@ -63,12 +68,14 @@ class Authentication extends Component {
         onboardingStep: 'connecting to server'
       })
     } finally {
-      this.setState({ fetching: false })
+      if (!this.unmounted) {
+        this.setState({ fetching: false })
```

Comment:
Yes it would be cleaner, I am aware of this pattern, it means that for every promises susceptible to call setState, you would store them into an internal array of the component ? and then in componentWIllUnmount, cancelAllInternalPromises ?

Reply:
Make sense to create an internal array for each component. Actually I just wrap every promise in a... promise. And add a `cancel` method to the wrapped promise. If the first promise resolves and the cancel method has been called before, we just `reject` the promise with 'canceled' string. 

Very simple implementation : https://github.com/cozy/cozy-drive/pull/1611/files#diff-ac9f6ecd36bdefb9a86d4bafe3c6a30c 


Actually we have to call .cancel() for every promise we want to cancel during unmounting the component, but having an array can be great :) 

(It seems like an hack and it is... at the beginning I wanted to really cancel the task but didn't find a way to do it with a promise )